### DISCUSSION TITLE
Is there any way to catch the current progress in python?

### DISCUSSION THREAD

**Discussion by maxi07** [ocrmypdf/OCRmyPDF#1234] [Q&A]
I am running some OCR processing using a bigger python app. At some point, I will run the OCR. Is there any way I can catch the current progress (eg page x/y completed) to show some progress to the user? The only thing I could think of was trying to reroute the logging, but maybe there is a better way?

**Comment by jbarlow83** [comment]
You could create a plugin that registers a custom progress bar and use this to capture progress.
https://ocrmypdf.readthedocs.io/en/latest/plugins.html
Not all events are logged in this way.
